sdk/eventhub/Azure.Messaging.EventHubs.Shared/tests/Processor/PartitionLoadBalancerTests.cs

```diff
@@ -0,0 +1,413 @@
+            const int NumberOfPartitions = 3;
+            var partitionIds = Enumerable.Range(1, NumberOfPartitions).Select(p => p.ToString()).ToArray();
+            var partitionManager = new MockCheckPointStorage((s) => Console.WriteLine(s));
+            var loadbalancer1 = new PartitionLoadBalancer(
+                partitionManager, Guid.NewGuid().ToString(), ConsumerGroup, FullyQualifiedNamespace, EventHubName, TimeSpan.FromMinutes(1));
+            var loadbalancer2 = new PartitionLoadBalancer(
+                partitionManager, Guid.NewGuid().ToString(), ConsumerGroup, FullyQualifiedNamespace, EventHubName, TimeSpan.FromMinutes(1));
+
+            // Ownership should start empty.
+
+            var completeOwnership = await partitionManager.ListOwnershipAsync(FullyQualifiedNamespace, EventHubName, ConsumerGroup);
+            Assert.That(completeOwnership.Count(), Is.EqualTo(0));
+
+            // Start the loadbalancer so that it claims a random partition until none are left.
+
+            for (int i = 0; i < NumberOfPartitions; i++)
+            {
+                await loadbalancer1.RunLoadBalancingAsync(partitionIds, tokenSource.Token);
+            }
+
+            completeOwnership = await partitionManager.ListOwnershipAsync(FullyQualifiedNamespace, EventHubName, ConsumerGroup);
+
+            // All partitions are owned by loadbalancer1.
+
+            Assert.That(completeOwnership.Count(p => p.OwnerIdentifier.Equals(loadbalancer1.Identifier)), Is.EqualTo(NumberOfPartitions));
+
+            // Stopping the loadbalancer should relinquish all partition ownership.
+
+            await loadbalancer1.RelinquishOwnershipAsync(tokenSource.Token);
+
+            completeOwnership = await partitionManager.ListOwnershipAsync(loadbalancer1.FullyQualifiedNamespace, loadbalancer1.EventHubName, loadbalancer1.ConsumerGroup);
+
+            // No partitions are owned by loadbalancer1.
+
+            Assert.That(completeOwnership.Count(p => p.OwnerIdentifier.Equals(loadbalancer1.Identifier)), Is.EqualTo(0));
+
+            // Start loadbalancer2 so that the loadbalancer claims a random partition until none are left.
+            // All partitions should be immediately claimable even though they were just claimed by the loadbalancer1.
+
+            for (int i = 0; i < NumberOfPartitions; i++)
+            {
+                await loadbalancer2.RunLoadBalancingAsync(partitionIds, tokenSource.Token);
+            }
+
+            completeOwnership = await partitionManager.ListOwnershipAsync(loadbalancer1.FullyQualifiedNamespace, loadbalancer1.EventHubName, loadbalancer1.ConsumerGroup);
+
+            // All partitions are owned by loadbalancer2.
+
+            Assert.That(completeOwnership.Count(p => p.OwnerIdentifier.Equals(loadbalancer2.Identifier)), Is.EqualTo(NumberOfPartitions));
+
+            await loadbalancer2.RelinquishOwnershipAsync(tokenSource.Token);
```

Comment:
nit: is there a point in relinquishing ownership at the end of the tests?

Reply:
nope, it just looked nice! 😄 
removed them.